src/app/clusters/general-commissioning-server/general-commissioning-server.cpp

```diff
@@ -175,42 +208,88 @@ bool emberAfGeneralCommissioningClusterArmFailSafeCallback(app::CommandHandler *
         // to allow commissioners the opportunity to obtain this failsafe for the purpose of commissioning
         if (!failSafeContext.IsFailSafeArmed() &&
             Server::GetInstance().GetCommissioningWindowManager().IsCommissioningWindowOpen() &&
-            commandObj->GetSubjectDescriptor().authMode == Access::AuthMode::kCase)
+            ctx.mCommandHandler.GetSubjectDescriptor().authMode == Access::AuthMode::kCase)
         {
+            ChipLogError(FailSafe, "FailSafe arming denied: commissioning window is open.");
             response.errorCode = CommissioningErrorEnum::kBusyWithOtherAdmin;
-            commandObj->AddResponse(commandPath, response);
         }
-        else if (commandData.expiryLengthSeconds == 0)
+        else if (req.expiryLengthSeconds == 0)
         {
             // Force the timer to expire immediately.
             failSafeContext.ForceFailSafeTimerExpiry();
             // Don't set the breadcrumb, since expiring the failsafe should
             // reset it anyway.
             response.errorCode = CommissioningErrorEnum::kOk;
-            commandObj->AddResponse(commandPath, response);
         }
         else
         {
-            CheckSuccess(
-                failSafeContext.ArmFailSafe(accessingFabricIndex, System::Clock::Seconds16(commandData.expiryLengthSeconds)),
-                Failure);
-            Breadcrumb::Set(commandPath.mEndpointId, commandData.breadcrumb);
+            CheckSuccess(failSafeContext.ArmFailSafe(accessingFabricIndex, System::Clock::Seconds16(req.expiryLengthSeconds)),
+                         Failure);
+            Breadcrumb::Set(ctx.mRequestPath.mEndpointId, req.breadcrumb);
             response.errorCode = CommissioningErrorEnum::kOk;
-            commandObj->AddResponse(commandPath, response);
         }
     }
     else
     {
+        ChipLogError(FailSafe, "FailSafe arming denied: busy with another admin or already armed.");
         response.errorCode = CommissioningErrorEnum::kBusyWithOtherAdmin;
-        commandObj->AddResponse(commandPath, response);
     }
 
-    return true;
+    ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+}
+
+void GeneralCommissioningAttrAccess::HandleSetRegulatoryConfig(HandlerContext & ctx,
```

Comment:
Why did this get moved?  Makes it really hard to review.  Unless there was a _really_ good reason for it, put it back where it was.  Changes requested for this part, because I have not reviewed this new impl.